presto-main/src/main/java/com/facebook/presto/execution/NodeTaskMap.java

```diff
@@ -49,12 +64,22 @@ public void addTask(InternalNode node, RemoteTask task)
 
     public int getPartitionedSplitsOnNode(InternalNode node)
     {
-        return createOrGetNodeTasks(node).getPartitionedSplitCount();
+        return (int) createOrGetNodeTasks(node).getPartitionedSplitCount();
```

Comment:
keep split count as `AtomicInt` maybe to prevent the unsafe cast to `int` ?

Reply:
TaskStatsTracker is using AtomicLong. This would require another class definition.